presto-main/src/main/java/com/facebook/presto/execution/resourceGroups/InternalResourceGroup.java

```diff
@@ -964,23 +1245,33 @@ public int hashCode()
         public RootInternalResourceGroup(
                 String name,
                 BiConsumer<InternalResourceGroup, Boolean> jmxExportListener,
-                Executor executor)
+                Executor executor,
+                Integer version)
         {
-            super(Optional.empty(), name, jmxExportListener, executor, true);
+            super(Optional.empty(), name, jmxExportListener, executor, true, version);
         }
 
-        public synchronized void processQueuedQueries()
+        public RootInternalResourceGroup(String name, BiConsumer<InternalResourceGroup, Boolean> jmxExportListener, Executor executor, InternalResourceGroup root, Integer version)
         {
-            internalRefreshStats();
-            while (internalStartNext()) {
-                // start all the queries we can
+            super(Optional.empty(), name, jmxExportListener, executor, root, true, version);
+        }
+
+        public void processQueuedQueries()
+        {
+            synchronized (getRoot()) {
+                internalRefreshStats();
+                while (internalStartNext()) {
+                    // start all the queries we can
+                }
             }
         }
 
-        public synchronized void generateCpuQuota(long elapsedSeconds)
+        public void generateCpuQuota(long elapsedSeconds)
         {
-            if (elapsedSeconds > 0) {
-                internalGenerateCpuQuota(elapsedSeconds);
+            synchronized (getRoot()) {
```

Comment:
Same here, what's the purpose of `getRoot()`?

Reply:
ditto